chef-config/lib/chef-config/fips.rb

```diff
@@ -0,0 +1,48 @@
+#
+# Author:: Matt Wrock (<matt@mattwrock.com>)
+# Copyright:: Copyright (c) 2016 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module ChefConfig
+
+  def self.fips?
+    if ChefConfig.windows?
+      require "win32/registry"
+      # from http://msdn.microsoft.com/en-us/library/windows/desktop/aa384129(v=vs.85).aspx
+      if ::RbConfig::CONFIG["target_cpu"] == "i386"
+        reg_type = Win32::Registry::KEY_READ | 0x100
+      elsif ::RbConfig::CONFIG["target_cpu"] == "x86_64"
+        reg_type = Win32::Registry::KEY_READ | 0x200
+      else
+        reg_type = Win32::Registry::KEY_READ
+      end
+
+      begin
+        Win32::Registry::HKEY_LOCAL_MACHINE.open('System\CurrentControlSet\Control\Lsa\FIPSAlgorithmPolicy', reg_type) do |policy|
+          policy["Enabled"] == 0 ? false : true
+        end
+      rescue Win32::Registry::Error
+        false
+      end
+    else
+      begin
+        File.read("/proc/sys/crypto/fips_enabled").chomp == "0" ? false : true
+      rescue Errno::ENOENT
+        false
+      end
```

Comment:
could probably be:

```
File.exist?("/proc/sys/crypto/fips_enabled") && File.read("/proc/sys/crypto/fips_enabled").chomp != "0"
```

avoid using the exception as normal flow control and invert the conditional...